xgcm/padding.py

```diff
@@ -0,0 +1,422 @@
+                            # TODO: Flip sign if vector is tangential
+
+                        # Tangential flip
+                        if swap_axis and not reverse:
+                            source_slice = source_slice.isel(
+                                {tangential_dim: slice(None, None, -1)}
+                            )
+                            # If the input is a tangential vector this flip needs
+                            # to be accompanied by a sign change
+                            if isvector:
+                                if vectoraxis != axname:
+                                    source_slice = -source_slice
+
+                        source_slice = source_slice.squeeze()
+
+                        # clean out everything on source_slice coordinates
+                        source_slice = source_slice.drop_vars(
+                            [co for co in source_slice.coords]
+                        )
+
+                        # Here I am trying to emulate the way xarray.pad deals with dimension coordinates
+                        # I will set them to nan in any case. This might change later.
+                        if target_dim in target_slice.coords:
+                            if (
+                                target_dim not in source_slice.dims
+                            ):  # in case this a 1 element padding slice
+                                source_slice = source_slice.expand_dims([target_dim])
+
+                            source_slice = source_slice.assign_coords(
+                                {
+                                    target_dim: np.full_like(
+                                        source_slice[target_dim].data,
+                                        np.nan,
+                                        dtype=float,  # Needed to properly convert int indicies to nan (super weird)
+                                    )
+                                }
+                            )
+
+                        # # FUCK THIS: FOR NOW BRUTE FORCE THIS STUFF
+                        # target_slice = _strip_all_coords(target_slice)
+                        # source_slice = _strip_all_coords(source_slice)
+
+                        # assemble the padded array
+                        if is_right:
+                            concat_list = [target_slice, source_slice]
+                        else:
+                            concat_list = [source_slice, target_slice]
+
+                        target_da = xr.concat(
+                            concat_list,
+                            dim=target_dim,
+                            coords="minimal",
+                            compat="override",
+                            join="override",
+                        )
+                        # TODO: Can we do this with an assignment in xarray? Maybe not important yet.
+        faces.append(target_da)
+
+    da_padded = xr.concat(faces, dim=facedim)
```

Comment:
Because each face is potentially padded on a different edge. I believe there is definitely a more efficient way to do this, but this was the easiest for me to track/develop.

Reply:
That makes sense as a first approach at least.